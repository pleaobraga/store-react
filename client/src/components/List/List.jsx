import React from 'react'
import PropTypes from 'prop-types'

import './List.scss'

export const List = ({ header, title, listItems, className }) => {
  return (
    <div className={`list ${className ? className : ''}`}>
      {title && <h1 className="list__title">{title}</h1>}
      <div className="list__header">{header}</div>
      {listItems}
    </div>
  )
}

List.defaultProps = {
  className: '',
  listItems: {}
}

List.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  header: PropTypes.node,
  listItems: PropTypes.node
}

export default React.memo(List)
