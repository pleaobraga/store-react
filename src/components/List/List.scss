@import '../../styles/colors';

.list {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: flex-start;

  &__title {
    font-size: 3.5rem;
    margin-bottom: 2.5rem;
  }

  &__header {
    display: flex;
    width: 100%;
    padding-bottom: 5px;

    * {
      color: $black;

      font-size: 2.5rem;
      font-weight: 500;

      padding-left: 1rem;
      padding-right: 1rem;
    }
  }
}
