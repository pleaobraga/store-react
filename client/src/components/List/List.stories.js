import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, number } from '@storybook/addon-knobs'
import { List } from './List'

export default storiesOf('Components | List', module).add(
  'default',
  () => (
    <List
      name={text('Name', 'Name')}
      quantity={number('Quantity', 2)}
      price={text('Price', '20,10')}
      currency={text('Currency', 'R$')}
    />
  ),
  { info: { inline: true, header: false } }
)
