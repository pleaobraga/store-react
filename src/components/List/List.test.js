import React from 'react'
import { shallow } from 'enzyme'
import List from './List'

describe('List', () => {
  const listItem = shallow(<List />)

  it('render proprely', () => {
    expect(listItem).toMatchSnapshot()
  })
})
