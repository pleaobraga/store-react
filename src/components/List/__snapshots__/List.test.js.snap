// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render proprely 1`] = `
<div
  className="list "
>
  <h1
    className="list__title"
  />
  <div
    className="list__header"
  />
</div>
`;
